### PR TITLE
Cldc 924 tenancy information burden reduction 2021 2022 form

### PR DESCRIPTION
**Pull request by dushan-madetech**
JIRA ticket can be found here https://digital.dclg.gov.uk/jira/browse/CLDC-924

**Start date for the log is in the 2021-2022 period**
Screenshot 2022-01-13 at 14 10 35

**New hint text for tenancy type if it is a starter tenancy**
Screenshot 2022-01-13 at 14 10 55

**Question for fixed term tenancy length**
Screenshot 2022-01-13 at 14 11 03

**New question about lettings in sheltered accommodation only for supported housing**
Screenshot 2022-01-13 at 14 11 16

**Check your answers page**
Screenshot 2022-01-13 at 14 11 54

**Check your answers page with more questions showing due to met conditions**
Screenshot 2022-01-13 at 14 18 18

### CODE REVIEW

File: app/models/case_log.rb [hunk 1]
```diff
@@ -48,6 +48,8 @@ class CaseLog < ApplicationRecord
   scope :for_organisation, ->(org) { where(owning_organisation: org).or(where(managing_organisation: org)) }
 
   enum status: { "not_started" => 0, "in_progress" => 1, "completed" => 2 }
+  enum letting_in_sheltered_accomodation: { "Yes - sheltered housing" => 0, "Yes - extra care housing" => 1, "No" => 2, "Don't know" => 3 }
```
Comment (baarkerlounger) [inline app/models/case_log.rb]: Worth moving both of these to the CaseLog constants file so they're all in the same place?

Comment (dushan-madetech) [reply app/models/case_log.rb]: Will do that now 😄

File: db/schema.rb [hunk 2]
```diff
@@ -162,9 +162,9 @@
     t.string "why_dont_you_know_la"
     t.integer "unitletas"
     t.integer "builtype"
+    t.datetime "property_void_date"
```
Comment (baarkerlounger) [inline db/schema.rb:165]: Do you get this when you run migrations on a fresh DB? If not should probably revert this.

**Review by baarkerlounger — APPROVED** [review]
LGTM

File: db/schema.rb (hunk 2 above)
Comment (dushan-madetech) [reply db/schema.rb:165]: Yep, I tried dropping and creating before applying migrations and would get the same result

File: app/models/case_log.rb (hunk 1 above)
Comment (dushan-madetech) [reply app/models/case_log.rb]: I've just updated this in efd1442406ba19dc6be8c35d6b905a1c4343a035, is that what you're expecting?

Comment (baarkerlounger) [reply app/models/case_log.rb]: Yep. Worth Paul reviewing but otherwise :ship:

File: config/forms/2021_2022.json [hunk 3]
```diff
@@ -1070,30 +1070,42 @@
...
+                  "check_answer_label": "Type of main tenancy",
+                  "header": "What is the type of tenancy?",
+                  "hint_text": "",
+                  "type": "radio",
+                  "answer_options": {
+                    "0": "Secure (including flexible)",
+                    "1": "Assured",
+                    "2": "Assured Shorthold",
+                    "3": "Licence agreement (almshouses only)",
+                    "4": "Other"
```
Comment (paulrobertlloyd) [inline config/forms/2021_2022.json]: Should be alphabetically ordered (with ‘Other’ last)

File: config/forms/2021_2022.json [hunk 4]
```diff
@@ -1070,30 +1070,42 @@
...
+            "tenancy_type_starter_text": {
               "header": "",
               "description": "",
               "questions": {
                 "tenancy": {
-                  "check_answer_label": "Type of main tenancy (after any starter/introductory period)",
+                  "check_answer_label": "Type of main tenancy after the starter period has ended?",
                   "header": "What is the type of tenancy after the starter period has ended?",
-                  "hint_text": "",
+                  "hint_text": "This is also known as an 'introductory period'.",
```
Comment (paulrobertlloyd) [inline config/forms/2021_2022.json]: 👁️

```suggestion
                  "hint_text": "This is also known as an ‘introductory period’.",
```

File: config/forms/2021_2022.json [hunk 5]
```diff
@@ -1111,23 +1123,43 @@
...
+                  "header": "Is this letting in sheltered accommodation?",
                   "hint_text": "",
                   "type": "radio",
                   "answer_options": {
-                    "0": "This landlord",
-                    "1": "Another registered provider - includes housing association or local authority"
+                    "0": "Yes - sheltered housing",
+                    "1": "Yes - extra care housing",
+                    "2": "No",
+                    "3": "Don't know"
```
Comment (paulrobertlloyd) [inline config/forms/2021_2022.json]: Add divider

File: config/forms/2021_2022.json [hunk 6]
```diff
@@ -1962,6 +1994,41 @@
...
+                    "0": "Weekly for 52 weeks",
+                    "1": "Fortnightly",
+                    "2": "Four-weekly",
+                    "3": "Calendar monthly",
+                    "4": "Weekly for 50 weeks",
+                    "5": "Weekly for 49 weeks",
+                    "6": "Weekly for 48 weeks",
+                    "7": "Weekly for 47 weeks",
+                    "8": "Weekly for 46 weeks",
+                    "9": "Weekly for 53 weeks"
```
Comment (paulrobertlloyd) [inline config/forms/2021_2022.json]: Order alphabetically

**Review by paulrobertlloyd — CHANGES_REQUESTED** [review]
Also… going by the images, the width of the key names in the summary lists looks too wide – incorrect markup?

File: config/forms/2021_2022.json (hunk 5 above)
Comment (dushan-madetech) [reply config/forms/2021_2022.json]: I'm not quite sure what you mean here, do you have an example?

File: config/forms/2021_2022.json (hunk 3 above)
Comment (dushan-madetech) [reply config/forms/2021_2022.json]: Got it, will fix this one 😄

File: config/forms/2021_2022.json (hunk 6 above)
Comment (dushan-madetech) [reply config/forms/2021_2022.json]: Will make sure this is fixed 😄

Comment (dushan-madetech) [reply config/forms/2021_2022.json]: Does this now look correct? @paulrobertlloyd 🙂

File: config/forms/2021_2022.json (hunk 3 above)
Comment (dushan-madetech) [reply config/forms/2021_2022.json]: This should be fixed now 😄

File: config/forms/2021_2022.json (hunk 5 above)
Comment (paulrobertlloyd) [reply config/forms/2021_2022.json]: I think it’s this… adds an ‘or’ between the two options

```
"2": "No",
"divider": true,
"3": "Don’t know"
```

File: config/forms/2021_2022.json (hunk 6 above)
Comment (paulrobertlloyd) [reply config/forms/2021_2022.json]: 👌

File: config/forms/2021_2022.json (hunk 5 above)
Comment (dushan-madetech) [reply config/forms/2021_2022.json]: Screenshot 2022-01-13 at 15 18 22

Is this what you had in mind?

Comment (paulrobertlloyd) [reply config/forms/2021_2022.json]: Yup! Thanks!

Comment (roobottom) [reply config/forms/2021_2022.json]: That LGTM

File: app/models/constants/case_log.rb [hunk 7]
```diff
@@ -1081,4 +1081,17 @@ module Constants::CaseLog
...
+    "not_started" => 0,
+    "in_progress" => 1,
+    "completed" => 2,
+  }.freeze
+
+  SHELTERED_ACCOMODATION = {
+    "Yes - sheltered housing" => 1,
+    "Yes - extra care housing" => 2,
+    "No" => 3,
+    "Don't know" => 4,
```
Comment (paulrobertlloyd) [inline app/models/constants/case_log.rb]: 👀 

```suggestion
    "Don’t know" => 4,
```

File: config/forms/2021_2022.json [hunk 8]
```diff
@@ -1111,23 +1123,44 @@
...
                   "hint_text": "",
                   "type": "radio",
                   "answer_options": {
-                    "0": "This landlord",
-                    "1": "Another registered provider - includes housing association or local authority"
+                    "0": "Yes - sheltered housing",
+                    "1": "Yes - extra care housing",
+                    "2": "No",
+                    "divider": true,
+                    "3": "Don't know"
```
Comment (paulrobertlloyd) [inline config/forms/2021_2022.json]:
```suggestion
                    "3": "Don’t know"
```

File: config/forms/2021_2022.json [hunk 9]
```diff
@@ -1111,23 +1123,44 @@
...
-              }
+              },
+              "depends_on": [{ "startertenancy": "Yes" }]
             },
-            "letting_provider": {
+            "fixed_term_tenancy": {
               "header": "",
               "description": "",
               "questions": {
-                "landlord": {
```
Comment (baarkerlounger) [inline config/forms/2021_2022.json:1120]: @dushan-madetech  @paulrobertlloyd  it looks like `landlord` dropped out of the form here without being replaced by a derived variable but it is in the data dictionary. That doesn't seem correct? Should it be derived?

**Pull request merged**